x/cdp/keeper/savings.go

```diff
@@ -27,37 +28,46 @@ func (k Keeper) DistributeSavingsRate(ctx sdk.Context, debtDenom string) error {
 	modAccountCoins := k.getModuleAccountCoins(ctx, dp.Denom)
 	totalSupplyLessModAccounts := k.supplyKeeper.GetSupply(ctx).GetTotal().Sub(modAccountCoins)
 	surplusDistributed := sdk.ZeroInt()
+
 	var iterationErr error
+	// TODO: avoid iterating over all the accounts by keeping the storing stable coin
+	// holders' addresses separately.
```

Comment:
+1 on leaving TODO here. I believe v0.39 of sdk will store balances such that you can iterate over only accounts that hold a balance of a particular coin type.